stable/openstack-base/README.md

```diff
@@ -1,217 +1,329 @@
 
-### Booting an instance
+    ssh-keygen -q -N '' -f ~/.ssh/id_mykey
 
-First generate a SSH keypair so that you can access your instances once you've booted them:
+To import a keypair:
 
-    mkdir -p ~/.ssh
-    touch ~/.ssh/id_rsa_cloud
-    chmod 600 ~/.ssh/id_rsa_cloud
-    openstack keypair create mykey > ~/.ssh/id_rsa_cloud
+    openstack keypair create --public-key ~/.ssh/id_mykey.pub mykey
 
-**Note:** you can also upload an existing public key to the cloud rather than generating a new one:
+## Configure security groups
 
-    openstack keypair create --public-key ~/.ssh/id_rsa.pub mykey
+To allow ICMP (ping) and SSH traffic to flow to cloud instances create
+corresponding rules for each security group:
 
-You can now boot an instance on your cloud:
+    for i in $(openstack security group list | awk '/default/{ print $2 }'); do
+        openstack security group rule create $i --protocol icmp --remote-ip 0.0.0.0/0;
+        openstack security group rule create $i --protocol tcp --remote-ip 0.0.0.0/0 --dst-port 22;
+	done
+
+You only need to perform this step once.
+
+## Create an instance
+
+To create a Bionic instance called 'bionic-1':
 
     openstack server create --image bionic --flavor m1.small --key-name mykey \
         --nic net-id=$(openstack network list | grep internal | awk '{ print $2 }') \
-        bionic-test
+        bionic-1
+
+## Attach a volume
+
+This step is optional.
+
+To create a 10GiB volume in Cinder and attach it to the new instance:
 
-### Attaching a volume
+    openstack volume create --size=10 <volume-name>
+    openstack server add volume bionic-1 <volume-name>
 
-First, create a 10G volume in cinder:
+The volume becomes immediately available to the instance. It will however need
+to be formatted and mounted before usage.
 
-    openstack volume create --size=10 <name-of-volume>
+## Assign a floating IP address
 
-then attach it to the instance we just booted:
+To request and assign a floating IP address:
 
-    openstack server add volume bionic-test <name-of-volume>
+    FLOATING_IP=$(openstack floating ip create -f value -c floating_ip_address ext_net)
+    openstack server add floating ip bionic-1 $FLOATING_IP
 
-The attached volume will be accessible once you login to the instance (see below).  It will need to be formatted and mounted!
+## Log in to an instance
 
-### Accessing your instance
+To log in to the new instance:
 
-In order to access the instance you just booted on the cloud, you'll need to assign a floating IP address to the instance:
+    ssh -i ~/.ssh/id_mykey ubuntu@$FLOATING_IP
 
-    openstack floating ip create ext_net
-    openstack server add floating ip bionic-test <new-floating-ip>
+## Access the cloud dashboard
 
-and then allow access via SSH (and ping) - you only need to do these steps once:
+The cloud dashboard (Horizon) can be accessed by going to:
 
-    openstack security group list
+http://<openstack-dashboard-ip>/horizon
```

Comment:
This renders incorrectly when viewing in GH:

http:///horizon

Reply:
Thank you Chris. :cookie: Fixes committed.